### PR TITLE
SC-1524

### PR DESCRIPTION
**Pull request by anmol2302**
**Ticket Ref:** [SC-1524](https://project-sunbird.atlassian.net/browse/SC-1524)
**Release:** 2.5.0
1: Added `user/v2/bulk/upload ` API for uploading shadow user's

### CODE REVIEW

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://539.qa.docs.sunbird.org/2.5.0/ .<br/> **Note:** Only _2.5.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://539.qa.docs.sunbird.org/2.5.0/apis/bulkupload// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://539.qa.docs.sunbird.org/2.5.0/ .<br/> **Note:** Only _2.5.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://539.qa.docs.sunbird.org/2.5.0/apis/bulkupload// <br />  <br/> **Modified sections are -** <br />

File: apis/bulkupload/bulkserviceapi.yaml [hunk 1]
```diff
@@ -75,6 +75,61 @@ paths:
...
+        '400':
+          description: 'BAD REQUEST. The "bulk upload user" operation failed ! The possible reason for failure is that you may have missed providing input for a mandatory parameter.'
+          schema:
+            $ref: '#/definitions/BulkUploadErrorResponse'
+        '500':
+          description: 'INTERNAL SERVER ERROR ! Looks like something went wrong! These errors are tracked automatically, but if the problem persists feel free to contact us. In the meantime, try refreshing.'
+  /user/v2/bulk/upload:
+    post:
+      description: >-
+        This API is associated with upload shadow users in bulk on the Sunbird Platform,shadow user are those who want to migrate from custodian org to non-custodian org.
```
Comment (manzarul) [inline apis/bulkupload/bulkserviceapi.yaml]: This API is associated with uploading shadow user on the Sunbird Platform.Only org admin role can upload user. If any uploaded user identifier (email,phone) matched with custodian user then he/she will be eligible for migration , from custodian to state user.

File: apis/bulkupload/bulkserviceapi.yaml [hunk 2]
```diff
@@ -75,6 +75,59 @@ paths:
...
+           - The endpoint for **Upload Users in Bulk** is `/user/v2/bulk/upload`
+           - The fields marked with an asterisk (*) are mandatory. They cannot be null or empty.
+      summary: Upload Shadow Users in Bulk
+      tags:
+        - Bulk Operation APIs
+      operationId: bulk upload shadow user req1
```
Comment (manzarul) [inline apis/bulkupload/bulkserviceapi.yaml]: bulk upload shadow user

File: apis/bulkupload/bulkserviceapi.yaml [hunk 3]
```diff
@@ -75,6 +75,59 @@ paths:
...
+        - name: user
+          in: formData
+          required: true
+          type: file
+          format: file
+          description: >-
+            This is a .csv file.
+              - Each line of the file is a data record. Each record consists of one or more user fields, separated by commas. Each successfully created user is associated with root org of uploader (org admin)
+            The CSV file format supported is mentioned below:
+              - NAME: Name of user(firstName)
```
Comment (manzarul) [inline apis/bulkupload/bulkserviceapi.yaml]: name will be your complete name (firstname lastname)

File: apis/bulkupload/bulkserviceapi.yaml [hunk 4]
```diff
@@ -75,6 +75,59 @@ paths:
...
+              - PHONE: Phone number. Either phone or email is mandatory.
+              - EMAIL: Email. Either phone or email is mandatory.
+              - Ext Org id: Organisation ID to which user must be associated(extOrgId).it is not mandatory attribute, if passed user will be assosiated to organisation(RootOrg and Sub-Org).if not then user will be only associated with the org admin root org
+              - Input Status: it is mandatory attribute. this attribute will decide the status(active/inactive) in an organization.
+              - Ext User ID: External ID of user. Either, school ID or external ID is mandatory for user association to organisation.
```
Comment (manzarul) [inline apis/bulkupload/bulkserviceapi.yaml]: External identity of user. This attribute is mandatory.

File: apis/bulkupload/bulkserviceapi.yaml [hunk 5]
```diff
@@ -75,6 +75,59 @@ paths:
...
+        - name: X-Authenticated-User-Token
+          in: header
+          required: true
+          type: string
+          description: >-
+            Access token of registered user performing given API request.
+
       responses:
         '200':
           description: 'OK ! Successful operation."bulk upload user" operation was successfully executed.'
```
Comment (manzarul) [inline apis/bulkupload/bulkserviceapi.yaml]: success with processid. This process id can be used for further communication.

File: apis/bulkupload/bulkserviceapi.yaml (hunk 1 above)
Comment (anmol2302) [reply apis/bulkupload/bulkserviceapi.yaml]: done

File: apis/bulkupload/bulkserviceapi.yaml (hunk 4 above)
Comment (anmol2302) [reply apis/bulkupload/bulkserviceapi.yaml]: done

File: apis/bulkupload/bulkserviceapi.yaml (hunk 5 above)
Comment (anmol2302) [reply apis/bulkupload/bulkserviceapi.yaml]: done

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://539.qa.docs.sunbird.org/2.5.0/ .<br/> **Note:** Only _2.5.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://539.qa.docs.sunbird.org/2.5.0/apis/bulkupload// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @anmol2302,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://539.qa.docs.sunbird.org/2.5.0/ .<br/> **Note:** Only _2.5.0_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://539.qa.docs.sunbird.org/2.5.0/apis/bulkupload// <br />  <br/> **Modified sections are -** <br />

**Pull request merged**